plugins/koterm.koplugin/terminputtext.lua

```diff
@@ -0,0 +1,391 @@
+--[[
+module used for terminal emulator to override InputText
+]]
+
+local InputText = require("ui/widget/inputtext")
+local UIManager = require("ui/uimanager")
+local dbg = require("dbg")
+local util = require("util")
+
+local esc_seq = {
+    backspace = "\008",
+    cursor_left =  "\027[D",
+    cursor_right = "\027[C",
+    cursor_up =    "\027[A",
+    cursor_down =  "\027[B",
+    cursor_pos1 =  "\027[7~",
+    cursor_end =   "\027[8~",
+    page_up =      "\027[5~",
+    page_down =    "\027[6~",
+}
+
+local TermInputText = InputText:extend{
+    terminal_mode = true,
+    maxr = 40,
+    maxc = 80,
+    strike_callback = nil,
+}
+
+function TermInputText:onTapTextBox(arg, ges)
+    -- disable positioning cursor by tap in emulator mode
+    if self.terminal_mode then
+        return true
+    end
+    return InputText.onTapTextBox(self, arg, ges)
+end
+
+function TermInputText:addChars(chars, skip_callback)
+    if not chars then
```

Comment:
You can add comments between the sections you just needed to copy as-is from InputText:
```
-- As done in InputText:

-- Specific TermInputText:
```
So it's easier to notice and update later if we have to fix/tweak InputText.